taxi/taxi_gym_env.py

```diff
@@ -0,0 +1,146 @@
+#importing required libraries for gym env
+import gym
+import numpy as np
+from visgrid.taxi.taxi import VisTaxi5x5
+
+
+#debugging
+import pdb
+
+
+'''Taxi Environment: inherits from gym.Env class'''
+class TaxiEnv(gym.Env):
+
+    metadata = {'render.modes': ['human', 'rgb_array']}
+
+
+    def __init__(self, num_passengers=1, grayscale =False, have_goal = True, randomize_positions = True, ignore_rewards = False, max_steps_per_episode = None):
+        '''creates taxi gym env: passed into GymWrapper for dreamerv2'''
+
+        #base TaxiEnv environment: logistics
+        self.taxi_env = VisTaxi5x5(n_passengers =  num_passengers, grayscale = grayscale)
+        taxi_rendering = self.taxi_env.reset(goal = have_goal, explore = randomize_positions)
+        taxi_rendering = self._convert_to_int(taxi_rendering)
+
+        #stored for future resets
+        self.have_goal = have_goal; self.randomize_positions = randomize_positions
+
+        #observation space for TaxiEnv: RGB array
+        #self.observation_space = gym.spaces.Box(low=0, high=255, shape = taxi_rendering.shape, dtype = np.uint8)
+        self.observation_space = taxi_rendering
```

Comment:
What's going on here? Shouldn't the observation_space be a gym `Space` instead of a numpy array?